src/clj/game/cards/programs.clj

```diff
@@ -2925,16 +2979,19 @@
                        :value 2}]})
 
 (defcard "Saci"
-  {:hosting {:card #(and (ice? %)
-                         (can-host? %))}
+  {:hosting {:req (req (and (ice? target)
+                            (installed? target)
+                            (can-host? state target)))}
    :events [{:event :rez
-             :req (req (and (same-card? (:card context) (:host card))
-                            (not= (:title (:card context)) "Magnet")))
+             :req (req (same-card? (:card context) (:host card)))
+             ;;(not= (:title (:card context)) "Magnet")))
              :msg "gain 3 [Credits]"
              :async true
              :effect (effect (gain-credits :runner eid 3))}
             {:event :derez
              :req (req (same-card? (:card context) (:host card)))
+             ;; special cheat for working with magnet
+             :while-disabled true
```

Comment:
Looking at the code again, I don't think this is necessary? It should be possible to update disabled status in `derez` and then this will just "magically" work.

Reply:
Yeah, just add `(update-disabled-cards state)` before the `trigger-event` call in `derez` and you can drop `:while-disabled` (including the other uses in `engine`).

Reply:
neat